### PR TITLE
Feat: V2 ENS support

### PR DESCRIPTION
**Pull request by colmharte**
Add support for ENS v2

fileLimit are now taken from exposureConfig rather than being passed in

Adds version number to backend url requests for better metric tracking

Summarised exposureWindow data is returned to backend when contact alerts

Adds support for contiguous mode

Exposure alerts now only fire if the contact date is more recent than previous contacts

### CODE REVIEW

File: android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        }
+
+        return exposureWindows;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private List<WindowData> extractExposureWindows(List<ExposureWindow> windows, Long daysSinceEpoch, ExposureConfig config) {
+        List<ExposureWindow> matchWindows = windows;
+
+        if (daysSinceEpoch != -1) {
```
Comment (AlanSl) [inline android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java]: A comment that this is if not contiguous mode here would make this a lot easier to follow

File: android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        });
+        scanData.setNumScans(windows.size());
+        return scanData;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private ScanData buildScanData(ExposureConfig config, List<ScanInstance> scanData) {
+        ScanData scanItem = new ScanData();
+        scanItem.setNumScans(scanData.size());
+        double[] thresholdWeightings = new double[]{config.getImmediateDurationWeight(), config.getNearDurationWeight(), config.getMediumDurationWeight(), config.getOtherDurationWeight()};
```
Comment (AlanSl) [inline android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java:152]: Just checking I've followed this correctly:

 - In V1 we expect `thresholdWeightings` as an array in config with decimal values 0-1
 - In V2 we instead expect individual named weightings in config which are also used as `attenuationWeightings` and are passed as 0-100

**Review by AlanSl — COMMENTED** [review]
I'm not in a position to do a full review but having read through nothing jumps out as looking wrong. Just a couple of comments to check I'm following it right.

File: android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java (hunk 2 above)
Comment (colmharte) [reply android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java:152]: Yes correct, its tied to changes Apple /  Google made in their configuration object

File: android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java (hunk 1 above)
Comment (colmharte) [reply android/src/main/java/ie/gov/tracing/nearby/riskcalculation/RiskCalculationV2.java]: Updated to use getContiguousMode instead

**Pull request merged**